From 705f0b7afa7bc1c67cd1c1a3ede881e5672f40f8 Mon Sep 17 00:00:00 2001
From: Hariprasad Kelam <hkelam@marvell.com>
Date: Tue, 20 Jul 2021 18:02:40 +0530
Subject: [PATCH 1649/1921] octeontx2-af: fix 50G ethtool mapping

OcteonTx2 supports 2lane 50G serdes mode and CN10K supports
single lane 50G serdes mode.

Fix serdes mode to ethtool link mode mapping for OceonTx2 50G links.
They are incorrect mapped to single lane ethtool modes.

Change-Id: Ieb1ecf84ebed218fb012755110729403dfbe8828
Signed-off-by: Hariprasad Kelam <hkelam@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/56975
Reviewed-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
Tested-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/cgx.c  | 16 ++++++++--------
 .../marvell/octeontx2/nic/otx2_ethtool.c         | 16 ++++++++--------
 2 files changed, 16 insertions(+), 16 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
index 2892d431d468..28b6c7ae5e27 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
@@ -1220,19 +1220,19 @@ static void otx2_map_ethtool_link_modes(u64 bitmask,
 	case  ETHTOOL_LINK_MODE_10000baseKX4_Full_BIT:
 		set_mod_args(args, 40000, 0, 0, BIT_ULL(CGX_MODE_40GAUI_C2C));
 		break;
-	case  ETHTOOL_LINK_MODE_50000baseSR_Full_BIT:
+	case  ETHTOOL_LINK_MODE_50000baseSR2_Full_BIT:
 		set_mod_args(args, 50000, 0, 0, BIT_ULL(CGX_MODE_50G_C2C));
 		break;
 	case  ETHTOOL_LINK_MODE_56000baseKR4_Full_BIT:
 		set_mod_args(args, 50000, 0, 0, BIT_ULL(CGX_MODE_50G_4_C2C));
 		break;
-	case  ETHTOOL_LINK_MODE_50000baseLR_ER_FR_Full_BIT:
+	case  ETHTOOL_LINK_MODE_50000baseDR_Full_BIT:
 		set_mod_args(args, 50000, 0, 0, BIT_ULL(CGX_MODE_50G_C2M));
 		break;
-	case  ETHTOOL_LINK_MODE_50000baseCR_Full_BIT:
+	case  ETHTOOL_LINK_MODE_50000baseCR2_Full_BIT:
 		set_mod_args(args, 50000, 0, 1, BIT_ULL(CGX_MODE_50G_CR));
 		break;
-	case  ETHTOOL_LINK_MODE_50000baseKR_Full_BIT:
+	case  ETHTOOL_LINK_MODE_50000baseKR2_Full_BIT:
 		set_mod_args(args, 50000, 0, 1, BIT_ULL(CGX_MODE_50G_KR));
 		break;
 	case  ETHTOOL_LINK_MODE_10000baseLRM_Full_BIT:
@@ -1250,19 +1250,19 @@ static void otx2_map_ethtool_link_modes(u64 bitmask,
 	case  ETHTOOL_LINK_MODE_100000baseKR4_Full_BIT:
 		set_mod_args(args, 100000, 0, 1, BIT_ULL(CGX_MODE_100G_KR4));
 		break;
-	case  ETHTOOL_LINK_MODE_50000baseSR2_Full_BIT:
+	case  ETHTOOL_LINK_MODE_50000baseSR_Full_BIT:
 		set_mod_args(args, 50000, 0, 0,
 			     BIT_ULL(CGX_MODE_LAUI_2_C2C_BIT));
 		break;
-	case  ETHTOOL_LINK_MODE_50000baseDR_Full_BIT:
+	case  ETHTOOL_LINK_MODE_50000baseLR_ER_FR_Full_BIT:
 		set_mod_args(args, 50000, 0, 0,
 			     BIT_ULL(CGX_MODE_LAUI_2_C2M_BIT));
 		break;
-	case  ETHTOOL_LINK_MODE_50000baseCR2_Full_BIT:
+	case  ETHTOOL_LINK_MODE_50000baseCR_Full_BIT:
 		set_mod_args(args, 50000, 0, 1,
 			     BIT_ULL(CGX_MODE_50GBASE_CR2_C_BIT));
 		break;
-	case  ETHTOOL_LINK_MODE_50000baseKR2_Full_BIT:
+	case  ETHTOOL_LINK_MODE_50000baseKR_Full_BIT:
 		set_mod_args(args, 50000, 0, 1,
 			     BIT_ULL(CGX_MODE_50GBASE_KR2_C_BIT));
 		break;
diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
index 943ec6f323eb..2d4eb25463f0 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
@@ -1134,20 +1134,20 @@ static void otx2_get_link_mode_info(u64 link_mode_bmap,
 		ETHTOOL_LINK_MODE_40000baseCR4_Full_BIT,
 		ETHTOOL_LINK_MODE_40000baseKR4_Full_BIT,
 		ETHTOOL_LINK_MODE_10000baseKX4_Full_BIT,
-		ETHTOOL_LINK_MODE_50000baseSR_Full_BIT,
+		ETHTOOL_LINK_MODE_50000baseSR2_Full_BIT,
+		ETHTOOL_LINK_MODE_50000baseDR_Full_BIT,
 		ETHTOOL_LINK_MODE_56000baseKR4_Full_BIT,
-		ETHTOOL_LINK_MODE_50000baseLR_ER_FR_Full_BIT,
-		ETHTOOL_LINK_MODE_50000baseCR_Full_BIT,
-		ETHTOOL_LINK_MODE_50000baseKR_Full_BIT,
+		ETHTOOL_LINK_MODE_50000baseCR2_Full_BIT,
+		ETHTOOL_LINK_MODE_50000baseKR2_Full_BIT,
 		ETHTOOL_LINK_MODE_10000baseLRM_Full_BIT,
 		ETHTOOL_LINK_MODE_100000baseSR4_Full_BIT,
 		ETHTOOL_LINK_MODE_100000baseLR4_ER4_Full_BIT,
 		ETHTOOL_LINK_MODE_100000baseCR4_Full_BIT,
 		ETHTOOL_LINK_MODE_100000baseKR4_Full_BIT,
-		ETHTOOL_LINK_MODE_50000baseSR2_Full_BIT,
-		ETHTOOL_LINK_MODE_50000baseDR_Full_BIT,
-		ETHTOOL_LINK_MODE_50000baseCR2_Full_BIT,
-		ETHTOOL_LINK_MODE_50000baseKR2_Full_BIT,
+		ETHTOOL_LINK_MODE_50000baseSR_Full_BIT,
+		ETHTOOL_LINK_MODE_50000baseLR_ER_FR_Full_BIT,
+		ETHTOOL_LINK_MODE_50000baseCR_Full_BIT,
+		ETHTOOL_LINK_MODE_50000baseKR_Full_BIT,
 		ETHTOOL_LINK_MODE_100000baseSR2_Full_BIT,
 		ETHTOOL_LINK_MODE_100000baseLR2_ER2_FR2_Full_BIT,
 		ETHTOOL_LINK_MODE_100000baseCR2_Full_BIT,
-- 
2.31.1

